File: internal/model/feed.go
package model

import "time"

// Feed はフィードのデータモデルです。
type Feed struct {
	ID             string    `json:"id"`
	Name           string    `json:"name" binding:"required"`
	URL            string    `json:"url" binding:"required,url"`
	PluginType     string    `json:"plugin_type" binding:"required"`
	FolderID       string    `json:"folder_id,omitempty"`
	UpdateInterval int       `json:"update_interval"`
	LastUpdated    time.Time `json:"last_updated,omitempty"`
	CreatedAt      time.Time `json:"created_at"`
}

File: frontend/src/services/feed.ts
import api from './api';

export interface Feed {
  id: string;
  name: string;
  url: string;
  plugin_type: string;
  folder_id?: string;
  update_interval: number;
  last_updated?: string;
  created_at: string;
}

export const feedService = {
  getAllFeeds: async (): Promise<Feed[]> => {
    const response = await api.get<Feed[]>('/feeds');
    return response.data;
  },

  getFeedById: async (id: string): Promise<Feed> => {
    const response = await api.get<Feed>(`/feeds/${id}`);
    return response.data;
  },

  createFeed: async (feed: Omit<Feed, 'id' | 'created_at' | 'last_updated'>): Promise<Feed> => {
    const response = await api.post<Feed>('/feeds', feed);
    return response.data;
  },

  updateFeed: async (id: string, feed: Omit<Feed, 'id' | 'created_at' | 'last_updated'>): Promise<Feed> => {
    const response = await api.put<Feed>(`/feeds/${id}`, feed);
    return response.data;
  },

  deleteFeed: async (id: string): Promise<void> => {
    await api.delete(`/feeds/${id}`);
  },
};
